app/test/testmerginapi.cpp

```diff
@@ -61,11 +62,59 @@ void TestMerginApi::testDownloadProject()
     ProjectList projects = mMerginProjectModel->projects();
     Q_ASSERT(!mMerginProjectModel->projects().isEmpty());
     mProjectName = mMerginProjectModel->projects().at(0)->name;
-    qDebug() << "TestMerginApi::testListProjectFinished PASSED";
+    qDebug() << "TestMerginApi::testDownloadProject PASSED";
+}
+
+void TestMerginApi::testCreateProject()
+{
+    QSignalSpy spy(mApi, SIGNAL(projectCreated()));
+    mProjectName = "TEMPORARY_TEST_PROJECT";
+    mApi->createProject(mProjectName);
+
+    QVERIFY(spy.wait(1000));
+    QCOMPARE(spy.count(), 1);
+
+    ProjectList projects = getProjectList();
+    Q_ASSERT(!mMerginProjectModel->projects().isEmpty());
+
+    bool containsTestProject = false;
+    for (std::shared_ptr<MerginProject> project: projects) {
+        if (project->name == mProjectName) containsTestProject = true;
+    }
+    Q_ASSERT(containsTestProject);
+    qDebug() << "TestMerginApi::testCreateProject PASSED";
+}
+
+void TestMerginApi::testDeleteProject()
+{
+    testCreateProject();
+
+    QSignalSpy spy(mApi, SIGNAL(serverProjectDeleted(QString)));
+    mProjectName = "TEMPORARY_TEST_PROJECT";
+    mApi->deleteProject(mProjectName);
+    spy.wait(1000);
+
+    ProjectList projects = getProjectList();
+    bool containsTestProject = false;
+    for (std::shared_ptr<MerginProject> project: projects) {
+        if (project->name == mProjectName) containsTestProject = true;
+    }
+    Q_ASSERT(!containsTestProject);
+    qDebug() << "TestMerginApi::testDeleteProject PASSED";
```

Comment:
Please add also standalone tests in separate methods to 1. try to create a project that already exists (should fail), 2. try to delete a project that does not exist (should fail).